contracts/0.8.9/StakingRouter.sol

```diff
@@ -0,0 +1,570 @@
+        address indexed actor,
+        StakingModuleStatus fromStatus,
+        StakingModuleStatus toStatus
+    );
+    event WithdrawalCredentialsSet(bytes32 withdrawalCredentials);
+    event ContractVersionSet(uint256 version);
+    /**
+     * Emitted when the StakingRouter received ETH
+     */
+    event StakingRouterETHReceived(uint256 amount);
+
+    /// @dev errors
+    error ErrorZeroAddress(string field);
+    error ErrorBaseVersion();
+    error ErrorValueOver100Percent(string field);
+    error ErrorStakingModuleStatusNotChanged();
+    error ErrorStakingModuleIsPaused();
+    error ErrorStakingModuleIsNotPaused();
+    error ErrorEmptyWithdrawalsCredentials();
+    error ErrorDirectETHTransfer();
+
+    struct StakingModuleCache {
+        address stakingModuleAddress;
+        uint16 moduleFee;
+        uint16 treasuryFee;
+        uint16 targetShare;
+        uint256 activeKeysCount;
+        uint256 availableKeysCount;
+    }
+
+    bytes32 public constant MANAGE_WITHDRAWAL_CREDENTIALS_ROLE = keccak256("MANAGE_WITHDRAWAL_CREDENTIALS_ROLE");
+    bytes32 public constant MODULE_PAUSE_ROLE = keccak256("MODULE_PAUSE_ROLE");
+    bytes32 public constant MODULE_RESUME_ROLE = keccak256("MODULE_RESUME_ROLE");
+    bytes32 public constant MODULE_MANAGE_ROLE = keccak256("MODULE_MANAGE_ROLE");
+    bytes32 public constant STAKING_ROUTER_DEPOSIT_ROLE = keccak256("STAKING_ROUTER_DEPOSIT_ROLE");
+
+    /// Version of the initialized contract data
+    /// NB: Contract versioning starts from 1.
+    /// The version stored in CONTRACT_VERSION_POSITION equals to
+    /// - 0 right after deployment when no initializer is invoked yet
+    /// - N after calling initialize() during deployment from scratch, where N is the current contract version
+    /// - N after upgrading contract from the previous version (after calling finalize_vN())
+    bytes32 internal constant CONTRACT_VERSION_POSITION = keccak256("lido.StakingRouter.contractVersion");
+
+    /// @dev Credentials which allows the DAO to withdraw Ether on the 2.0 side
+    bytes32 internal constant WITHDRAWAL_CREDENTIALS_POSITION = keccak256("lido.StakingRouter.withdrawalCredentials");
+
+    uint256 public constant DEPOSIT_SIZE = 32 ether;
+
+    uint256 public constant TOTAL_BASIS_POINTS = 10000;
+
+    /// @dev total count of staking modules
+    uint256 private _stakingModulesCount;
+
+    /// @dev id of the last added staking module. This counter grow on staking modules adding
+    uint24 private _lastStakingModuleId;
+
+    /// @dev mapping is used instead of array to allow to extend the StakingModule
+    mapping(uint256 => StakingModule) private _stakingModules;
+
+    /// @dev Position of the staking modules in the `_stakingModules` map, plus 1 because
+    ///      index 0 means a value is not in the set.
+    mapping(uint24 => uint256) private _stakingModuleIndicesOneBased;
+
+    constructor(address _depositContract) BeaconChainDepositor(_depositContract) {}
```

Comment:
Petrify impl (init with 0) right in the constructor and set version to 1?

Reply:
only version locked in constructor, it should be enough since other values are zeros by default